src/axolotl/prompt_strategies/vicuna.py

```diff
@@ -0,0 +1,176 @@
+"""
+Prompt Strategy for finetuning Vicuna v1.1+ models
+see also https://huggingface.co/psmathur/orca_mini_v2_7b for more information
+
+Use dataset type: "vicuna" in conig.yml to use this prompt style.
+
+E.g. in the config.yml:
+```
+datasets:
+  - path: vicuna_finetune_train.jsonl
+    type: vicuna
+```
+
+The dataset itself should look like this:
+```
+{'conversation':[{"from": "human", "value": "Who are you?"}, {"from": "gpt", "value": "I am Vicuna"},...]}
+```
+in a jsonl file. The first message should be from the human, the second from gpt.
+For a custom system message, the first "from" can be "system" (followed by "human" and "gpt" turns).
+"""
+
+import logging
+from dataclasses import dataclass, field
+from typing import Generator, List, Sequence
+
+from axolotl.prompt_tokenizers import PromptTokenizingStrategy
+from axolotl.prompters import IGNORE_TOKEN_ID
+
+
+@dataclass
+class VicunaConversation:
+    """A class that manages prompt templates and keeps all conversation history.
+    copied from https://github.com/lm-sys/FastChat/blob/main/fastchat/conversation.py"""
+
+    name: str = "vicuna_v1.1"
+    # The system prompt
+    system: str = (
+        "A chat between a curious user and an artificial intelligence assistant. "
+        "The assistant gives helpful, detailed, and polite answers to the user's questions."
+    )
+    roles: Sequence[str] = ("USER", "ASSISTANT")
+    messages: List[List[str]] = field(default_factory=list)
+    offset: int = 0
+    sep: str = " "
+    sep2: str = "</s>"
+
+    def get_prompt(self) -> str:
+        """Get the prompt for generation."""
+        seps = [self.sep, self.sep2]
+        ret = self.system + seps[0]
+        for i, (role, message) in enumerate(self.messages):
+            if (i == len(self.messages) - 1) and (role == self.roles[0]):
+                # last message is from user (due to length),
+                #  return prompt without it
+                return ret
+            if message:
+                ret += role + ": " + message + seps[i % 2]
+            else:
+                ret += role + ":"
+        return ret
+
+    def append_message(self, role: str, message: str):
+        """Append a new message."""
+        self.messages.append([role, message])
+
+
+class VicunaTokenizingStrategy(PromptTokenizingStrategy):
+    """
+    Tokenizing strategy for ShareGPT prompts.
+    adapted from https://github.com/lm-sys/FastChat/blob/main/fastchat/train/train.py
+    """
+
+    def tokenize_prompt(self, prompt):
+        conv = next(self.prompter.build_prompt(prompt))
+        conversation_str = conv.get_prompt()
+
+        # Tokenize conversations
+        input_ids = self.tokenizer(
+            conversation_str,
+            return_tensors="pt",
+            padding="max_length",
+            max_length=self.tokenizer.model_max_length,
```

Comment:
```suggestion
            max_length=self.sequence_len,
```

This is problematic for llamav2 which has some randomly high number. Please let me know if this breaks any intended feature. Following other tokenizing code here, we need also set it follow seq len.

```json
"model_max_length": 1000000000000000019884624838656,
```